flight.py

```diff
@@ -0,0 +1,225 @@
+    parser.add_argument('-d', '--debug', action="store_true")
+    parser.add_argument('routine', nargs="?", default=None, type=str)
+
+    args = parser.parse_args()
+
+    if not args.debug and not args.routine:
+        logging.error("Please specify routine.")
+        sys.exit(1)
+    return args
+
+
+def create_routes(args, controller):
+    """
+
+    Constructs all routes necessary for network flight
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Arguments assembled by argparse.
+    controller : flight.drone.drone_controller.DroneController
+        The second parameter.
+
+    Returns
+    -------
+    list
+        Blueprints to be added to Flask app
+
+    """
+    pass
+
+
+def debug_add_task(controller, command, meta):
+    if "priority" in meta:
+        priority = get_priority(meta["priority"])
+    if "direction" in meta:
+        direction = get_direction(meta["direction"])
+
+    if command == "exit":
+        controller.add_exit_task(c.Priorities.HIGH)
+        return False
+    elif command == "land":
+        controller.add_land_task(priority)
+    elif command == "hover":
+        controller.add_hover_task(fc.DEFAULT_ALTITUDE, int(meta["time"]),
+                                  priority)
+    elif command == "takeoff":
+        controller.add_takeoff_task(int(meta["altitude"]))
+    elif command == "move":
+        controller.add_linear_movement_task(direction, int(meta["time"]),
+                                            priority)
+    return True
+
+
+def get_priority(priority):
+    return get_enum(c.Priorities, priority)
+
+
+def get_direction(direction):
+    return get_enum(c.Directions, direction)
+
+
+def get_enum(enum, key):
+    item = key.upper()
+    if hasattr(enum, item):
+        return getattr(enum, item)
+    else:
+        raise InvalidDirectionException(
+            'Expected value for type {}, got {}.'.format(type(enum), item))
+
+
+def start_ai(controller, routine):
+    importlib.import_module("flight.AIs.{}".format(routine))
+
+
+def kill_ai(controller):
+    pass
+
+
+def list_tasks(controller):
+    return controller.queue
+
+
+def add_task(args, controller, data):
+    if args.debug:
+        # If in debug mode
+        if not ("command" in data and "meta" in data):
+            logging.error("command and meta required")
+            return True
+        command = data["command"].lower()
+        meta = data["meta"]
+        try:
+            return debug_add_task(controller, command, meta)
+        except (InvalidDirectionException, InvalidPriorityException) as err:
+            return err
+        except Exception as err:
+            logging.error("Unexpected error, killing drone" + str(err))
+            debug_add_task(controller, "exit", {})
+            return False
+    elif args.routine:
+        # If in production mode, starting or stopping drone is only option
+        # get push data and check if start or kill
```

Comment:
Correct me if I'm wrong here, but is this comment supposed to say "get post data" instead of "get push data"?